cou/steps/analyze.py

```diff
@@ -78,6 +79,32 @@ def _split_control_plane_and_data_plane(
                 control_plane.append(app)
         return control_plane, data_plane
 
+    @classmethod
+    def _remove_data_plane_apps(
```

Comment:
As stated in another comment, I think we should merge this function into `_split_control_plane_and_data_plane`. But if in the end we decide to keep them separated, we should at least update the function title. Right now it is misleading: it sounds like you are deleting data plane apps where in fact you are postponing the upgrade of some control-plane components.

Reply:
I changed its name to `_move_to_data_plane_if_required`. But I do not agree with merging.